### ISSUE TITLE
DK/Spell Death Grip shoud intrerupt npc cast too

### ISSUE THREAD

**Issue by CraftedRO** [TrinityCore/TrinityCore#26137]
**Description:**

Death Knight spell [Death Grip](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=49576) dosen't intrerupt npc-s / mobs .

**Expected behaviour:**

Death Knight spell [Death Grip](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=49576) shoud intrerupt npc-s / mobs too like it does on players .

> [Gripping a target will interrupt any cast currently in progress. This can be used to great effect in PvP, interrupting healers or preventing node captures. Casting will be interrupted even if the target is barely moved, such as when engaging a caster at point-blank range.](https://wow.gamepedia.com/Death_Grip)

**Steps to reproduce the problem:**

1. For a quick reproduction make a Death Knight level 80 , .tele icecrown and go to Sindragosa's Fall - Icecrown , there you'll find some [Frostbrood Whelp](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=31718)
2. Use [Death Grip](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=49576) on them when they cast [Frost Blast](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?spell=60814).
3. Notice they won't be intrerupted .

**Branch(es):**

3.3.5

**TC rev. hash/commit:** e2ba73e9a35d

TrinityCore rev. e2ba73e9a35d 2021-02-23 02:17:09 +0200 (3.3.5 branch) (Win64, Release, Static)

**Operating system:** 

Windows 10 x64

https://user-images.githubusercontent.com/24683355/108895204-f9148c00-761b-11eb-9bff-a1cece9cbf76.mp4

**Comment by CraftedRO** [comment]
<details>
  <summary>400aef35a19f9a8f8b11d741cecaf9a4ac1d7d42 still valid, just tried smth like this, it works but for some reason most of the time it break the pull effect not sure why</summary>


```c++

    void HandleDummy(SpellEffIndex /*effIndex*/)
    {
        if (Unit* target = GetHitUnit())
        {
            if (target->HasUnitState(UNIT_STATE_CASTING))
                target->CastStop(); // Stop casting

            else if (!target->HasAuraType(SPELL_AURA_DEFLECT_SPELLS)) // Deterrence
                target->CastSpell(GetExplTargetDest()->GetPosition(), GetEffectValue(), true);
        }
    }
```
</details>

<details>
  <summary>
Here's a video recorded on cata ptr 4.4.0.54670, this is how shoud work:</summary>


https://github.com/TrinityCore/TrinityCore/assets/24683355/85ee5248-71d2-4d9c-a0f6-b5df45618aeb


</details>

**Comment by Aokromes** [comment]
yeah a lot of times if you use death grip over a mob that is casting and ain't rooted pull fails.

**Comment by Shauren** [comment]
Unconditionally interrupting is wrong - it should respect interrupt flags (not all npc spells are interrupted on movement)

**Comment by CraftedRO** [comment]
well yeah my code is more likely just a proof of concept that can be enhaced by someone that know what other checks need to be added, one can be not to check for players since there already works probably smth like && target->GetTypeId() =! TYPEID_PLAYER) or whatever

**Comment by Aokromes** [comment]
one good sample of spell that can be interrupted but aint interrupted pulling is https://aowow.trinitycore.info/?npc=29822 lava brust, on cata ptr it's interrupted.

**Comment by CraftedRO** [comment]
maybe related https://github.com/TrinityCore/TrinityCore/issues/18416
